app/assets/javascripts/discourse/app/components/sidebar/filter.gjs

```diff
@@ -12,6 +12,11 @@ import { bind } from "discourse-common/utils/decorators";
 export default class Filter extends Component {
   @service sidebarState;
 
+  constructor() {
+    super(...arguments);
+    document.documentElement.addEventListener("keydown", this.handleKeydown);
```

Comment:
Ah I think this might not be the best way to do it; you already use `on "input"` for the filter itself:

https://github.com/discourse/discourse/blob/385d7962c2272a5c467ae9bd6e36311e3d0a6c74/app/assets/javascripts/discourse/app/components/sidebar/filter.gjs#L56-L61

I think you can add `on "keydown"` and do your `handleKeydown` escape handling and filter clearing there.

Reply:
It was kind of by design ;) My thinking was that you could click around and still hit "Escape" to reset the filter. With  keydown solution, input has to be active.

Not a big deal, let's start with the cleaner solution and we can always expand it if needed :)

Reply:
Ah in that case you may want to use `KeyboardShortcuts.addShortcut`, but yes let's just start out with active input only